kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -24,42 +24,100 @@
 namespace Kratos
 {
 	CalculateDiscontinuousDistanceToSkinProcess::CalculateDiscontinuousDistanceToSkinProcess(ModelPart& rVolumePart, ModelPart& rSkinPart)
-	: FindIntersectedGeometricalObjectsProcess(rVolumePart, rSkinPart) {
-		
+		: mrSkinPart(rSkinPart), mrVolumePart(rVolumePart), mFindIntersectedObjectsProcess(rVolumePart, rSkinPart)
+	{
+	}
+
+	CalculateDiscontinuousDistanceToSkinProcess::~CalculateDiscontinuousDistanceToSkinProcess()
+	{
 	}
 
-	CalculateDiscontinuousDistanceToSkinProcess::~CalculateDiscontinuousDistanceToSkinProcess(){}
+	void CalculateDiscontinuousDistanceToSkinProcess::Initialize()
+	{
+		// Initialize the intersected objects process
+		mFindIntersectedObjectsProcess.Initialize();
 
+		// Reset the nodal distance values
+		const double initial_distance = 1.0;
```

Comment:
I have noticed that you have not access to the bounding box here and It is not so recommendable to calculate it again here. So I would leave it as it is until giving a bounding box interface to find intersections process